From c5c2087feccdc7a3d0f15059c5196b8718cd6b6f Mon Sep 17 00:00:00 2001
From: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
Date: Thu, 8 Oct 2020 08:57:39 +0200
Subject: [PATCH 3/5] Add image directory creation to avoid error to access
 recents directory

Trace: E TaskPersister: File error accessing recents directory (directory doesn't exist?).

Signed-off-by: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
Change-Id: Ic039636cb0fda3781edba78b208e11a92c3f4f9b
---
 .../core/java/com/android/server/wm/TaskPersister.java   | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/wm/TaskPersister.java b/services/core/java/com/android/server/wm/TaskPersister.java
index 20af250280f..da76dce2b9d 100644
--- a/services/core/java/com/android/server/wm/TaskPersister.java
+++ b/services/core/java/com/android/server/wm/TaskPersister.java
@@ -496,7 +496,14 @@ public class TaskPersister implements PersisterQueue.Listener {
     }
 
     static File getUserImagesDir(int userId) {
-        return new File(Environment.getDataSystemCeDirectory(userId), IMAGES_DIRNAME);
+        File imageDir = new File(Environment.getDataSystemCeDirectory(userId), IMAGES_DIRNAME);
+
+        if (! imageDir.exists()) {
+            // create recent_images directory to avoid error logs
+            imageDir.mkdirs();
+        }
+
+        return imageDir;
     }
 
     private static boolean createParentDirectory(String filePath) {
-- 
2.17.1

